File: src/external-services/ExternalServices/FT - Base/Models/RequestModificarDatosUsuario.cs
using Newtonsoft.Json;

namespace FTBase.Models
{
    /// <summary>
    /// Modelo de datos para modificar los datos de un usuario existente.
    /// Contiene los parámetros necesarios para actualizar la información personal de un usuario.
    /// </summary>
    public class RequestModificarDatosUsuario
    {
        /// <summary>
        /// Obtiene o establece el nombre de usuario.
        /// </summary>
        public string NombreUsuario { get; set; } = string.Empty;

        /// <summary>
        /// Obtiene o establece el nombre del usuario.
        /// </summary>
        public string Nombre { get; set; } = string.Empty;

        /// <summary>
        /// Obtiene o establece el primer apellido del usuario.
        /// </summary>
        public string PrimerApellido { get; set; } = string.Empty;

        /// <summary>
        /// Obtiene o establece el segundo apellido del usuario.
        /// </summary>
        public string SegundoApellido { get; set; } = string.Empty;

        /// <summary>
        /// Obtiene o establece la URL de la imagen del usuario.
        /// </summary>
        public string Imagen { get; set; } = string.Empty;

        /// <summary>
        /// Obtiene o establece el objetivo de peso del usuario en kilogramos.
        /// </summary>
        public double ObjetivoPeso { get; set; } = 0.0;

        /// <summary>
        /// Obtiene o establece la fecha de nacimiento del usuario.
        /// </summary>
        public DateOnly FechaDeNacimiento { get; set; } = DateOnly.FromDateTime(DateTime.Now.AddYears(-18));

        /// <summary>
        /// Obtiene o establece la altura del usuario en centímetros.
        /// </summary>
        public double Altura { get; set; } = 0.0;

        /// <summary>
        /// Obtiene o establece el peso actual del usuario en kilogramos.
        /// </summary>
        public double Peso { get; set; } = 0.0;

        /// <summary>
        /// Obtiene o establece el sexo del usuario.
        /// </summary>
        public string Sexo { get; set; } = string.Empty;
    }
}

File: src/backend/FT.Account/spring-boot/FitnessTrackerServidor/src/main/java/com/fitness/aplicacion/dto/RequestModificarDatosUsuario.java
package com.fitness.aplicacion.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;


@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class RequestModificarDatosUsuario {
    @JsonProperty(value = "correo_electronico")
    private String email;

    @JsonProperty(value = "nombre_usuario")
    private String nombreUsuario;

    @JsonProperty(value = "nombre")
    private String nombre;

    @JsonProperty(value = "primer_apellido")
    private String primerApellido;

    @JsonProperty(value = "segundo_apellido")
    private String segundoApellido;

    @JsonProperty(value = "fecha_nacimiento")
    private LocalDate fechaDeNacimiento;

    // En centímetros
    @JsonProperty(value = "altura")
    private float altura;

    @JsonProperty(value = "imagen")
    private String imagen;

    @JsonProperty(value = "objetivo_peso")
    private float objetivoPeso;

    // En kilogramos
    @JsonProperty(value = "peso")
    private float peso;

    @JsonProperty(value = "sexo")
    private String sexo;
}
